src/Extensions/ServiceFabricGuest.Abstractions/options/ServiceFabricRestClientOptions.cs

```diff
@@ -1,19 +1,52 @@
-﻿// Copyright (c) Microsoft Corporation. All rights reserved.
-// Licensed under the MIT license.
-
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
 using System.ComponentModel.DataAnnotations;
-
+using System.Linq;
+using Microsoft.Omex.Extensions.Abstractions;
+
 namespace Microsoft.Omex.Extensions.ServiceFabricGuest.Abstractions
 {
 	/// <summary>
 	/// Option for the REST communication with cluster
 	/// </summary>
 	public class ServiceFabricRestClientOptions
-	{
+	{
 		/// <summary>
-		/// Settings definition for Cluster Endpoint
+		/// Settings definition for Cluster Endpoint Port
+		/// Value taken from https://docs.microsoft.com/en-us/azure/service-fabric/service-fabric-visualizing-your-cluster
+		/// </summary>
+		public int ClusterEndpointPort { get; set; } = 19080;
+
+		/// <summary>
+		/// Settings definition for Cluster Endpoint FQDN
 		/// </summary>
 		[Required(AllowEmptyStrings = false)]
-		public string ClusterEndpoint { get; set; } = string.Empty;
+		public string ClusterEndpointFQDN { get; set; } = DefaultFQDN();
+
+
+		/// <summary>
+		/// Settings definition for Cluster Endpoint 
+		/// </summary>
+		public string ClusterEndpoint => string.Format("http://{0}:{1}", ClusterEndpointFQDN, ClusterEndpointPort.ToString());
+
+		internal static string DefaultFQDN()
+		{
+			string? runtimeAddress = SfConfigurationProvider.GetSfVariable(RuntimeConnectionAddressEvnVariableName);
+			if (runtimeAddress == null)
+			{
+				return string.Empty;
+			}
+
+			string[] parts = runtimeAddress.Split(':');
+			if(parts.Length < 2)
+			{
+				return string.Empty;
+			}
+
+			return string.Join(":", parts.Take(parts.Length - 1));

```

Comment:
I think the default value that we return if we are failed to parse environment variables should be `localhost` instead of an empty string.

Also, the fact that we had to parse `Fabric_RuntimeConnectionAddress` to remove port kind of push me back to the idea of using `Fabric_NodeIPOrFQDN` is it possible that Node FQDN won't match Cluster FQDN? (Also it's not SF internal variable).

Reply:
Yes, it is possible that they don't match each other:
```
["Fabric_RuntimeConnectionAddress"] "localhost:19036" System.Collections.KeyValuePairs 
["Fabric_NodeIPOrFQDN"] "machine.name.something.com"
```